ts/test/session/unit/utils/job_runner/user_sync_job/UserSyncJob_test.ts

```diff
@@ -0,0 +1,408 @@
+import { expect } from 'chai';
+import { omit } from 'lodash';
+import Long from 'long';
+import Sinon from 'sinon';
+import { getSodiumNode } from '../../../../../../node/sodiumNode';
+import { NotEmptyArrayOfBatchResults } from '../../../../../../session/apis/snode_api/SnodeRequestTypes';
+import {
+  SnodeNamespaces,
+  SnodeNamespacesUserConfig,
+} from '../../../../../../session/apis/snode_api/namespaces';
+import { ConvoHub } from '../../../../../../session/conversations';
+import { LibSodiumWrappers } from '../../../../../../session/crypto';
+import { MessageSender } from '../../../../../../session/sending';
+import { UserUtils } from '../../../../../../session/utils';
+import { RunJobResult } from '../../../../../../session/utils/job_runners/PersistedJob';
+import { UserSync } from '../../../../../../session/utils/job_runners/jobs/UserSyncJob';
+import {
+  LibSessionUtil,
+  PendingChangesForUs,
+  UserDestinationChanges,
+  UserSuccessfulChange,
+} from '../../../../../../session/utils/libsession/libsession_utils';
+import { UserGenericWrapperActions } from '../../../../../../webworker/workers/browser/libsession_worker_interface';
+import { TestUtils } from '../../../../../test-utils';
+import { TypedStub, stubConfigDumpData } from '../../../../../test-utils/utils';
+import { NetworkTime } from '../../../../../../util/NetworkTime';
+
+function userChange(
+  sodium: LibSodiumWrappers,
+  namespace: SnodeNamespacesUserConfig,
+  seqno: number
+): PendingChangesForUs {
+  return {
+    ciphertext: sodium.randombytes_buf(120),
+    namespace,
+    seqno: Long.fromNumber(seqno),
+  };
+}
+
+describe('UserSyncJob run()', () => {
+  afterEach(() => {
+    Sinon.restore();
+  });
+  it('throws if no user keys', async () => {
+    const job = new UserSync.UserSyncJob({});
+
+    const func = async () => job.run();
```

Comment:
would be good to have the same comment here as the group sync job

```tsx
// Note: the run() function should never throw, at most it should return "permanent failure"
```

Reply:
@Bilb missed

Reply:
got it